Searching/BinarySearch.cpp

```diff
@@ -2,6 +2,12 @@
     Binary search:
     A search algorithm that finds the position of a target value
     within a sorted array
+
+    Time complexity:
+    O(log n), where n is the number of values in the vector
+
+    Space complexity:
+    O(log n), where n is the number of values in the vector
```

Comment:
Use uppercase _N_ (it's just a convention widely followed). Also remove the space between `log` and `N`.

I suggest writing it as _O(log(N))_.